### PR TITLE
[MB-1778] Fix subscription filtering and refresh button

### PR DESCRIPTION
**Pull request by sirojan24**
Add all nodes search in clustered mode and avoid the duplicated entries in the connected node drop down and fix the message count refresh button.
Associated JIRA Link : https://wso2.org/jira/browse/MB-1778

### CODE REVIEW

File: components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java [hunk 1]
```diff
@@ -93,6 +93,41 @@ public static SubscriptionManagementBeans getInstance(){
...
+                    parameters, signature);
+
+            if (result != null) {
+                messageCount = (long) result;
+            }
+            return messageCount;
+
+        } catch (MalformedObjectNameException | InstanceNotFoundException | MBeanException | ReflectionException e) {
+            throw new SubscriptionManagerException("Error while invoking mBean operations to get " +
+                    "message count for a subscription", e);
```
Comment (a5anka) [inline components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java]: Sentence should start with `+`

Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java]: Corrected in the latest commit

File: components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java [hunk 2]
```diff
@@ -93,6 +93,41 @@ public static SubscriptionManagementBeans getInstance(){
...
+                            "name=SubscriptionManagementInformation");
+
+            Object[] parameters = new Object[]{queueName};
+            String[] signature = new String[]{String.class.getName()};
+
+            Object result = mBeanServer.invoke(objectName,
+                    SubscriptionManagementConstants.PENDING_MESSAGE_COUNT_MBEAN_ATTRIBUTE,
+                    parameters, signature);
+
+            if (result != null) {
```
Comment (sdkottegoda) [inline components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java]: Let's flip the null check

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp [hunk 3]
```diff
@@ -115,7 +118,12 @@
             try {
                 client = new ClusterManagerClient(configContext, serverURL, cookie);
                 isClusteringEnabled = client.isClusteringEnabled();
-                allClusterNodeAddresses = client.getAllClusterNodeAddresses();
+                allClusterNodeAddressesInDropdown = client.getAllClusterNodeAddresses();
+                if(isClusteringEnabled){
```
Comment (sdkottegoda) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: A space needs to be there

**Comment by sajinidesilva** [comment]
Please check services.xml. in nthe current implementation even to browse topic subscriptions need to have queue permission. If so you need to change that.

File: components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java [hunk 4]
```diff
@@ -93,6 +93,41 @@ public static SubscriptionManagementBeans getInstance(){
...
+     * @throws SubscriptionManagerException
+     */
+    public long getPendingMessageCount(String queueName) throws SubscriptionManagerException{
+
+        long messageCount = 0;
+
+        MBeanServer mBeanServer = ManagementFactory.getPlatformMBeanServer();
+        try {
+            ObjectName objectName =
+                    new ObjectName("org.wso2.andes:type=SubscriptionManagementInformation," +
```
Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java]: Corrected in https://github.com/wso2/carbon-business-messaging/pull/357

File: components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java (hunk 2 above)
Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java]: Corrected in https://github.com/wso2/carbon-business-messaging/pull/357

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp (hunk 3 above)
Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: Corrected in https://github.com/wso2/carbon-business-messaging/pull/357

**Pull request merged**